xarray/tests/test_duck_array_wrapping.py

```diff
@@ -0,0 +1,373 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import xarray as xr
+
+# TODO: how to test these in CI?
```

Comment:
note: I just noticed [xarray-array-testing](https://github.com/xarray-contrib/xarray-array-testing)

Reply:
we do install pint in CI if that helps.

Reply:
Cupy requries a GPU, but we could add `jax` into Xarray's CI suite in GitHub actions, at least.

Reply:
the only thing we'll want to avoid is testing compat with too many array types in `xarray`'s CI. I guess I'll need to make progress on `xarray-array-testing` to avoid that, though.

Reply:
`jax` would be pretty easy to add. The CPU jaxlib package is only like 90MB. And it's a nice test case since it very closely matches the numpy API, but only implements `__array_namespace__`, and not `__array_function__` or `__array_ufunc__`.

Reply:
I agree, we should add JAX CPU. See https://github.com/pydata/xarray/tree/main/ci/requirements for the config files. Feel free to do this in this PR or a follow-up.